File: android/OpenGLDemo/app/src/main/java/com/yq/opengldemo/ui/MainActivity.java
package com.yq.opengldemo.ui;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.yq.opengldemo.R;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.btn_first_opengl).setOnClickListener(this);
        findViewById(R.id.btn_air_hockey).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_first_opengl:
                startActivity(FirstOpenGLProjectActivity.class);
                break;
            case R.id.btn_air_hockey:
                startActivity(AirHockeyActivity.class);
                break;
        }
    }

    private void startActivity(Class clz) {
        Intent intent = new Intent(this, clz);
        startActivity(intent);
    }
}

File: android/ARouterDemo/app/src/main/java/com/my/arouterdemo/activity/MainActivity.kt
package com.my.arouterdemo.activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import com.alibaba.android.arouter.launcher.ARouter
import com.my.arouterdemo.R
import com.my.common.Const
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_same_module_act.setOnClickListener(this)
        btn_in_module_act.setOnClickListener(this)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        Log.i(TAG, "onActivityResult, requestCode: $requestCode, resultCode: $resultCode")
    }

    override fun onClick(v: View?) {
        when (v) {
            //页面跳转，只需要在目标act上面加@Router注解
            btn_same_module_act -> {
                ARouter.getInstance()
                    .build(Const.PATH_SECOND_ACT)
                    //传递参数
                    .withString(SecondActivity.KEY_STR, "shit")
//                    .navigation()
                    //类似startActivityForResult
                    .navigation(this, REQUEST_CODE_SECOND)
            }
            //不同模块的act跳转和同模块的一样
            btn_in_module_act -> {
                ARouter.getInstance()
                    .build(Const.PATH_IN_MODULE_ACT)
                    .navigation()
            }
        }
    }

    companion object {
        const val TAG = "MainActivity"
        const val REQUEST_CODE_SECOND = 100
    }
}